digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/lane/LaneFiller.scala

```diff
@@ -313,13 +313,13 @@ class LaneFiller {
 
     val pieces = pointsOfInterest.zip(pointsOfInterest.tail)
     val segmentPieces = pieces.flatMap(p => SegmentPieceCreation(p._1, p._2, lanesZipped))
-                              .groupBy(_.laneCode)
+                              .groupBy(_.laneId)
 
-    val segmentsAndOrphanPieces = segmentPieces.map(n => extendOrDivide(n._2, lanesZipped.find(_.laneCode == n._1).get))
+    val segmentsAndOrphanPieces = segmentPieces.map(n => extendOrDivide(n._2, lanesZipped.find(_.id == n._1).get))
     val combinedSegment = segmentsAndOrphanPieces.keys.toSeq
     val newSegments = combinedSegment.flatMap(sl => generateLimitsForOrphanSegments(sl, segmentsAndOrphanPieces.getOrElse(sl, Seq()).sortBy(_.startM)))
 
-    val changedSideCodes = combinedSegment.filter(cl => lanesZipped.exists(sl => sl.id == cl.id && !sl.sideCode.equals(cl.sideCode) && sl.laneCode == cl.laneCode))
+    val changedSideCodes = combinedSegment.filter(cl => lanesZipped.exists(sl => sl.id == cl.id && !sl.sideCode.equals(cl.sideCode)))
                                            .map(sl => SideCodeAdjustment(sl.id, SideCode(sl.sideCode)))
```

Comment:
removing this 
sl.laneCode == cl.laneCode

Wil not create issues with lanes that are in different lane codes?

Reply:
That part of code was rollbacked from a previous attempt of fixing a bug that led to another bug unrelated to the issue you posted.That issue you are talking about wasn't and isn't happening.

Reply:
My concern is if this remove will not make that old bug appear again.
But if you say it is solved, it is solved.